bip-rpob-tx-template.mediawiki

```diff
@@ -0,0 +1,284 @@
+
+In choosing to standardise via the BIP peer-reviewed process, we hope to give the community greater confidence in making use of bitcoin-based revocable proof of burn technology.
+
+====Notes:====
+<references group="Note" />
+
+==Design==
+
+===Conceptual===
+''"Proof-of-Burn"'' transactions have long exited in Bitcoin, ''"OP_RETURN"'' transactions, (named after the op-code ''"return"'', that causes the public key script to return in failure, hence making the output unspendable and "burning" any value allocated to that output), these transactions have been actively mined since the release of Bitcoin-Core 0.9.
+
+Since OP_RETURN doesn't allow the public key script to be spent (as is it's design), there hasn't been any built-in solution addressing the requirements a application-useable proof-of-burn transaction, in particular:
+
+*A standard association with a public key in for transaction.
+*A standard protocol to publicly revoke a proof-of-burn transaction.
+
+In this proposal we satisfy these requirements by specifying a taproot public key in the op_return payload and an another output for revocation.
+
+===Other Requirements===
+Operation under <abbr title="Simple Payment Verification">SPV</abbr> assumptions:
+*Transactions are small: ''RPoB transactions (non-witness-part) are fixed to two possible sizes, depending if there is a change output, or not.''
+*Revocations are quickly and cheaply indexed: ''RPoB revocations only require obtaining the preimage to a double-sha256 hash. This is easily indexed and propagated in a gossip network.''
+
+Ease of implementation:
+*Use of only Taproot and Schnorr: ''To reduce the scope of the transaction, we use only Taproot for inputs and Outputs, and Schnorr for the signature that covers the funding outpoint.''
+
+Transaction funded by 3rd party:
+* It should be secure for an untrusted 3rd party fund the proof-of-burn transaction. ''Allowing for services to fund these transactions and be compensated with some sort of off-chain payment.''
+
+Future extendability:
+*Making use of the transaction version feature of bitcoin. ''This allows for easier indexing strategies in the future, (with a simple consensus change: to include in the coinbase the number of the transactions in the block by their transaction version).''
+* Including a byte for 8 version flag bits. ''This allows for a reliable way to upgrade in the future in a forward-comparable way.''
+
+===Transaction Design Overview:===
+[[File:bip-rpob-tx-template/rpob-template-summary-table.svg]]
+
+==Specification==
+This section specifies our RPoB transaction template.
+
+===Transaction===
+General form:
+<pre>
+version:      0x3
+marker:       0x0
+flag:         0x1
+txin_count:   0x1
```

Comment:
Why only a single input per transaction?

Reply:
Hello @luke-jr, thank you for your initial review.

A transaction that has many inputs could be quite large; it is a requirement that this design is optimized for SPV performance. - Hence the restriction.

---

I have updated and expanded the 'Other Requirements' section: https://github.com/veleslavs/bips/blob/bip-rpob-tx-template/bip-rpob-tx-template.mediawiki#other-requirements


> Transactions are tiny and simple: The RPoB transactions template constrains the rules of the transaction to the minimum. Both size are complexity limitations are used: Single Input, used to limit the size of the RPoB transactions; Single Optional Change Output, used to limit the size of the RPoB transactions; Exclusive use of Taproot, used to limit the complexity of the RPoB transactions.